### DISCUSSION TITLE
Is there a built-in UI solution for validation errors?

### DISCUSSION THREAD

**Discussion by ValYouW** [react-querybuilder/react-querybuilder#743] [Q&A] [answered]
Hi,

I am looking for an easy way to hint the user there are errors, I wondered if there is a built-in UI for that. I know I can just copy the CSS from the "Use validation" option in the demo but before I do that I wanted to see I am not missing something (like a `showErrors` prop or something).

Also, how exactly the validation error "reasons" array is being used? say I return some "reasons" how the user gets to see it?

Thanks.

**Comment by jakeboone02** [comment, marked as answer]
No, there is no built-in UI or styling for validation. The `reasons` array is not used by the default components at all, it's just a mechanism for custom components to display information to the user or trigger different styles, etc.

**Comment by ValYouW** [comment]
Ok thx .
